py/redrock/utils.py

```diff
@@ -255,6 +255,11 @@ def transmission_Lyman(zObj,lObs, use_gpu=False):
     min_wave = 0
     if (np.isscalar(zObj)):
         #zObj is a float
+        min_wave = lObs.min()/(1+zObj)
+        if (min_wave > Lyman_series['Lya']['line']):
+            #Return None if wavelength range doesn't overlap with Lyman series
+            #No need to perform any calculations in this case
+            return None
```

Comment:
@craigwarner-ufastro did this optimization come from you?  How significant an improvement was it?  To me it seems more convenient to always return the transmission, even when it is all ones, so that the end user doesn't have to check the return type every time.  Or perhaps this optimization could become an opt-in option, to be used internal to the code, but without imposing it on other users of transmission_Lyman.

Reply:
@sbailey yes this was from me - and if I recall it was a very significant difference because most non-QSO templates don't overlap with the Lyman series.  So having to create an array of all ones and multiply thousands of times for each target caused a not insignificant and unnecessary slowdown.  We could do the opt-in option though without hurting any performance.

Reply:
Done - new argument is `always_return_array`